djhtml/__main__.py

```diff
@@ -51,12 +57,31 @@ def main():
             _error(e)
             continue
 
+        # Guess tabwidth
+        if not options.tabwidth:
+            diff = 0
+            probabilities = [0] * 9
+            for line in source.split("\n"):
```

Comment:
I'm wondering how this deals with a pathological case like

```
<div>
    <div>
         <div></div>

         <div></div>

         <div></div>

         <div></div>

         <div></div>

         <div></div>
    </div>
</div>
```

On line 2 it does `probabilities[4] += 1`.

On line 3 it does `probabilities[4] += 1`

On line 4 the count is 0, so it doesn't do anything.

On line 5 the count is 8, so it does `probabilities[8] += 1`, and similarly on line 7, 9, 11, 13, resulting on `probabilities[8] = 5`.

On line 14 it does `probabilities[4] += 1`, and the same on line 15.

Result: it favours indent 8 over indent 4.

Reply:
I thought I solved this before, but evidently I hadn't. I have now added an `if line and not line.isspace()` statement and it passed the pathology test :+1: